src/components/Questionnaire/structured/handlers.ts

```diff
@@ -195,10 +204,51 @@ export const structuredHandlers: {
       }));
     },
   },
+  consent: {
+    getRequests: async (consents, { patientId }) =>
+      consents.map((consent) => ({
+        url: `/api/v1/patient/${patientId}/consent/`,
+        method: "POST",
+        body: {
+          ...consent,
+          file: undefined as unknown as File,
+        },
+        reference_id: "consent",
+      })),
+    afterRequest: async (data, res) => {
+      const consentId = (res.data as ConsentModel).id;
+      console.log(data.files);
+      data.files.forEach(async (file) => {
+        console.log(file);
+        const base64 = (await readFileAsDataURL(file.file)).split(",")[1];
+        await query(fileApi.upload, {
+          body: {
+            file_data: base64,
+            original_name: file.file.name,
+            name: file.name,
+            associating_id: consentId,
+            file_type: "consent",
+            file_category: "consent_attachment",
+          },
+        })({ signal: new AbortController().signal });
```

Comment:
calling `query` directly is a bad idea. `query` will throw for any non 2xx responses.

Reply:
Any other ideas? Encased it in a try catch for now